packages/client/src/comp/states/Base.tsx

```diff
@@ -23,75 +28,235 @@ type Props<Info> = React.PropsWithChildren<{
   const { voter } = VoterContainer.useContainer()
   const { fetchingData, setFetchingData } = FetchingDataContainer.useContainer()
+  const {
+    fields,
+    updateField,
+    canCheckRegistration,
+    validInputs,
+  } = ValidationContainer.useContainer()
+  const [ registrationStatus, setRegistrationStatus ] = React.useState<ExtendedStatus>(null)
 
-  const nameRef = useControlRef<Input>()
+  const firstNameRef = useControlRef<Input>()
+  const lastNameRef = useControlRef<Input>()
   const birthdateRef = useControlRef<Input>()
   const emailRef = useControlRef<Input>()
-  const phoneRef = useControlRef<Input>()
+  const telephoneRef = useControlRef<Input>()
   const mailingRef = useControlRef<Input>()
+
   if (!locale || !isImplementedLocale(locale) || !contact) return null
 
   const uspsAddress = address ? address.fullAddr : null
   const { city, county, otherCities, latLong } = locale
 
-  async function handleSubmit(event: React.FormEvent<HTMLFormElement>) {
-    event.persist()  // allow async function call
-    event.preventDefault()
-
+  const stateInfo = (forRegistrationStatus: boolean): StateInfo | null => {
     if (!address || !uspsAddress || !contact) {
       toast.error('Please fill all the required fields')
-      return
+      return null
     }
 
-    const baseInfo: StatelessInfo = {
+    const base: StatelessInfo = {
       city: contact.city ?? city,
       county: contact.county ?? county,
       otherCities,
       latLong,
       oid,
-      name: nameRef.value() || '',
-      birthdate: birthdateRef.value() || '',
-      email: emailRef.value() || '',
-      mailingAddress: mailingRef.value() || '',
-      phone: phoneRef.value() || '',
+      name: `${firstNameRef.value()} ${lastNameRef.value()}`,
+      birthdate: birthdateRef.value() ?? '',
+      email: emailRef.value() ?? '',
+      mailingAddress: mailingRef.value() ?? '',
+      phone: telephoneRef.value() ?? '',
       uspsAddress,
       contact,
       address,
     }
 
-    const info = enrichValues(baseInfo)
+    const info = enrichValues(
+      // Small hack to prevent SignatureBase toasting missing signatures
+      !forRegistrationStatus ? base : {...base, signature: 'skip'}
+    )
+
     if (!info) {
       // Do not dismiss previous errors which may give more details on bad fields
       toast.error('Please fill all the required fields in the right formats')
-      return
     }
+
+    return info
+  }
+
+  async function aboutRegistrationBlur(
+    id: 'firstName' | 'lastName' | 'birthdate',
+    e: React.ChangeEvent<HTMLInputElement>,
+  ) {
+    e.preventDefault()
+    e.persist() // allow async function call
+    updateField(id, e.currentTarget.value)
+    if (canCheckRegistration()) {
+      if (registrationStatus !== 'Ignored' && registrationStatus !== 'Active') {
+        const info = stateInfo(true)
+        if (!info) return
+        setFetchingData(true)
+
+        const result = await client.isRegistered(info)
```

Comment:
So let's have `isRegistered` accepted `RegisteredArgs` which are 

firstName,
lastName,
birthdate,
address,
city,
state,
zip,

You will also want to remove `stateInfoToParams` from the server since you can just use the values that you need.